addons/t9n/models/resource.py

```diff
@@ -14,3 +19,52 @@ class Resource(models.Model):
     project_id = fields.Many2one(
         comodel_name="t9n.project",
     )
+
+    _sql_constraints = [
+        (
+            "file_name_unique",
+            "unique(file_name)",
+            "A file with the same name already exists!"
+        ),
+    ]
+
+    def _decode_resource_file(self, resource_file):
+        try:
+            file_content = base64.b64decode(resource_file.get("file")).decode()
+            po_obj = polib.pofile(file_content)
+        except (IOError, UnicodeDecodeError):
+            raise UserError(_("This file: %s should be a .po file with a valid syntax." % resource_file.get("file_name")))
```

Comment:
Small English fix (the same fix applies for plural "files" as well).
Also some good practices with text to be translated :wink:: 
- use a named placeholder so the translators have a better idea of what the value will be
- use the `_()` variable injection correctly [see doc](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#use-translation-method-correctly)

```suggestion
            raise UserError(_("The file: %(file_name) should be a .po file with a valid syntax.", file_name=resource_file.get("file_name")))
```

Reply:
Thank youuu